app/controllers/shopping/addresses_controller.rb

```diff
@@ -58,8 +58,7 @@ def update
   end
 
   def select_address
-    address = current_user.addresses.find(params[:id])
-    update_order_address_id(address.id)
+    update_order_address_id(params[:id])
```

Comment:
I have not looked at how `update_order_address` works, but there is one advantage to the `current_user` version: It ensures that the address referred to by `params[:id]` belongs to the current user. The other implementation allows any ID, no matter who owns it.